### PR TITLE
VTGate Expression Evaluation

### PR DESCRIPTION
**Pull request by saifalharthi**
This PR adds the ability to evaluate expression at the VTGate level. At this point it handles Literals and bind variables.

### CODE REVIEW

**Comment by saifalharthi** [comment]
@systay @harshit-gangal

**Comment by systay** [comment]
Test failures do not seem to be related

**Comment by harshit-gangal** [comment]
--- FAIL: TestDatabaseFunc (0.00s)
##[error]    database_func_test.go:38: select:
        [[INT64(0)]] want
        [[VARCHAR("ks")]]

This might be related.

**Comment by saifalharthi** [comment]
Failing test is not related to the chagnes.

File: go/vt/vtgate/engine/projection.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		for _, exp := range p.Exprs {
+			result, err := exp.Evaluate(env)
+			if err != nil {
+				return nil, err
+			}
+			row = append(row, result.Value())
+		}
+		rows = append(rows, row)
+	}
+	result.Rows = rows
```
Comment (harshit-gangal) [inline go/vt/vtgate/engine/projection.go:53]: Does this need result from input.execute? The results are manipulated here.

**Comment by saifalharthi** [comment]
Flaky tests are back

**Comment by saifalharthi** [comment]
It's OK to merge as the failing tests are not related to the changes.

**Pull request merged**